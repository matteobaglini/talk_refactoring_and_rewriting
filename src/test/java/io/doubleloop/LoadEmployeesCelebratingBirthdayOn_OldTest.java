package io.doubleloop;

import static org.assertj.core.api.Assertions.assertThat;

// REFACTORING: introduce contract tests

public class LoadEmployeesCelebratingBirthdayOn_OldTest extends LoadEmployeesCelebratingBirthdayOnContractTest {

  @Override
  protected LoadEmployeesCelebratingBirthdayOnContract createLoadEmployeesCelebratingBirthdayOn(String filePath, String today) {
    return new LoadEmployeesCelebratingBirthdayOn_Old(filePath, date(today));
  }

}
