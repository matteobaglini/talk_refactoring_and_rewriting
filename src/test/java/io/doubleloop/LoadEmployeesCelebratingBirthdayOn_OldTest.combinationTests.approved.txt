[test-data-bs/employee_data.txt, 2024/10/08] => [Employee[firstName=John, lastName=Doe, birthDate=1982-10-08, email=john.doe@foobar.com], Employee[firstName=Mario, lastName=Rossi, birthDate=2000-10-08, email=mario.rossi@foobar.com]] 
[test-data-bs/employee_data.txt, 2024/10/09] => [] 
[test-data-bs/employee_data.txt, 2024/03/11] => [Employee[firstName=Mary, lastName=Ann, birthDate=1975-03-11, email=mary.ann@foobar.com]] 
